### ISSUE TITLE
Develop an IBM Websphere ClasspathResolver

### ISSUE THREAD

**Issue by admtriggo** [CruxFramework/crux#111]
_From [brunodep...@gmail.com](https://code.google.com/u/108972312674998482139/) on May 20, 2010 00:09:59_

Purpose of enhancement IBM WS is one of the most used Application Servers in Java (mainly in bank
and finance institutions, perhaps because of its easy integration with
mainframe, OpenMVS, CTG+CICS, etc).

I tested Crux on my machine, a Windows XP Pro SP3 BR, in the following
scenarios (always testing war and exploded war deploy):

Apache Tomcat 6.0.18: OK!
Apache Geronimo 2.1.4 with Jetty: OK! (see issues `#109` and `#110`).
IBM Websphere Community Edition (geronimo): OK! (see above note)
IBM Websphre 7.1 (only tested the war through the ibm console deploy tool):
Error attached (ev001.JPG file). Probably because of the ClasspathResolver.
Following the log output of this test:

[20/05/10 00:05:41:562 ART] 00000027 WebApp        A   SRVE0181I:
[HelloWorldApp_war#HelloWorldApp.war] [/HelloWorldApp] [Servlet.LOG]:
remoteServiceServlet: An IncompatibleRemoteServiceException was thrown
while processing this call.:
com.google.gwt.user.client.rpc.IncompatibleRemoteServiceException: Error
creating service
br.eti.kinoshita.crux.sample.client.remote.GreetingService. Cause: null
    at
br.com.sysmap.crux.core.server.dispatch.RemoteServiceServlet.getServiceForRequest(RemoteServiceServlet.java:179)
    at
br.com.sysmap.crux.core.server.dispatch.RemoteServiceServlet.processCall(RemoteServiceServlet.java:54)
    at
com.google.gwt.user.server.rpc.RemoteServiceServlet.processPost(RemoteServiceServlet.java:224)
    at
com.google.gwt.user.server.rpc.AbstractRemoteServiceServlet.doPost(AbstractRemoteServiceServlet.java:62)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:763)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:856)
    at
com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:966)
    at
com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:478)
    at
com.ibm.ws.wswebcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:463)
    at
com.ibm.ws.webcontainer.servlet.CacheServletWrapper.handleRequest(CacheServletWrapper.java:92)
    at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:744)
    at
com.ibm.ws.wswebcontainer.WebContainer.handleRequest(WebContainer.java:1425)
    at com.ibm.ws.webcontainer.channel.WCChannelLink.ready(WCChannelLink.java:92)
    at
com.ibm.ws.http.channel.inbound.impl.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:465)
    at
com.ibm.ws.http.channel.inbound.impl.HttpInboundLink.handleNewInformation(HttpInboundLink.java:394)
    at
com.ibm.ws.http.channel.inbound.impl.HttpICLReadCallback.complete(HttpICLReadCallback.java:102)
    at
com.ibm.ws.tcp.channel.impl.AioReadCompletionListener.futureCompleted(AioReadCompletionListener.java:152)
    at
com.ibm.io.async.AbstractAsyncFuture.invokeCallback(AbstractAsyncFuture.java:213)
    at
com.ibm.io.async.AbstractAsyncFuture.fireCompletionActions(AbstractAsyncFuture.java:195)
    at com.ibm.io.async.AsyncFuture.completed(AsyncFuture.java:136)
    at com.ibm.io.async.ResultHandler.complete(ResultHandler.java:193)
    at
com.ibm.io.async.ResultHandler.runEventProcessingLoop(ResultHandler.java:725)
    at com.ibm.io.async.ResultHandler$2.run(ResultHandler.java:847)
    at com.ibm.ws.util.ThreadPool$Worker.run(ThreadPool.java:1498)
Caused by: java.lang.RuntimeException: Error creating service
br.eti.kinoshita.crux.sample.client.remote.GreetingService. Cause: null
    at
br.com.sysmap.crux.core.server.dispatch.ServiceFactoryImpl.getService(ServiceFactoryImpl.java:37)
    at
br.com.sysmap.crux.core.server.dispatch.RemoteServiceServlet.getServiceForRequest(RemoteServiceServlet.java:162)
    ... 23 more
Caused by: java.lang.NullPointerException
    at
br.com.sysmap.crux.core.server.dispatch.ServiceFactoryImpl.getService(ServiceFactoryImpl.java:33)
    ... 24 more

It would be great have IBM WS support in Crux.

Cheers

**Attachment:** [ev001.JPG](http://code.google.com/p/crux-framework/issues/detail?id=111)

_Original issue: http://code.google.com/p/crux-framework/issues/detail?id=111_

**Comment by admtriggo** [comment]
_From [brunodep...@gmail.com](https://code.google.com/u/108972312674998482139/) on May 22, 2010 05:06:01_

Hi everybody.

The Crux WeblogicClassPathResolver of the UserManual works with IBM Websphere too!!!

I installed crux app's war through admin console. It creates a directory in
installedApps/. This directory is the exploded form of the war. 

The only way of deploying an exploded war that I'm aware of in IBM WAS, is
overwriting the contents of this directory. So, first you deploy an application
through the admin console and then hotdeploy it again as many times as you wish. 

So, I skipped this step, and am considering Crux is fully supported in IBM WAS ;-)

Cheers, 
B

**Attachment:** [crux_on_ibm.JPG](http://code.google.com/p/crux-framework/issues/detail?id=111#hc2)
